File: Ruby/problem4.rb
min = 800
max = 999

def palindrome?(x) 
  x.to_s == x.to_s.reverse
end

largest = 0

(min..max).each do |x|
  (min..max).each do |y|
    square = x*y
    largest = square if palindrome? square
  end
end

puts largest

File: Clojure/problem4.clj
(def problem-set (range 100 1000))

(defn reverse-num [num]
	(loop [x num rev 0]
		(if (= x 0)
			rev
			(recur (quot x 10) (+ (rem x 10) (* rev 10))))))

(defn pal? [x]
	(= x (reverse-num x)))

(defn largest-palindrome []
	(apply max
		(for [x problem-set 
					y problem-set
					:let [z (* x y)]
					:when (pal? z)]
			z)))

(println (largest-palindrome))